python/ray/serve/_private/proxy.py

```diff
@@ -1039,19 +1039,24 @@ async def send_request_to_replica(
         finally:
             if not proxy_asgi_receive_task.done():
                 proxy_asgi_receive_task.cancel()
-            else:
-                # If the server disconnects, status_code is set above from the
-                # disconnect message. Otherwise the disconnect code comes from
-                # a client message via the receive interface.
-                if (
-                    status is None
-                    and proxy_request.request_type == "websocket"
-                    and proxy_asgi_receive_task.exception() is None
-                ):
+
+            # If the server disconnects, status_code can be set above from the
+            # disconnect message. but it is not guaranteed.
+            # If client disconnects, the disconnect code comes from
+            # a client message via the receive interface, but is is not guaranteed.
+            if status is None and proxy_request.request_type == "websocket":
+                if not proxy_asgi_receive_task.cancelled:
+                    # The disconnect message is sent from the client.
                     status = ResponseStatus(
                         code=str(proxy_asgi_receive_task.result()),
                         is_error=True,
                     )
+                else:
+                    # The server disconnect without sending a disconnect message.
+                    status = ResponseStatus(
+                        code="1000",  # [Sihan] is there a better code for this?
```

Comment:
I think we can just reuse this `DISCONNECT_ERROR_CODE` here. Or make something more descriptive like "SERVER_DISCONNECTED" or something

Reply:
we'd better to follow practice to indicate the status of the websocket request in the proxy log. https://kapeli.com/cheat_sheets/WebSocket_Status_Codes.docset/Contents/Resources/Documents/index

Reply:
oh didn't know that's a thing for websockets XD

Reply:
please refer to the ASGI spec any time you aren't sure about response codes / other HTTP details:
https://asgi.readthedocs.io/en/latest/specs/main.html#:~:text=ASGI%20decomposes%20protocols%20into%20a,the%20corresponding%20event%20messages%20back.